docs/src/content/docs/guides/sidebar.mdx

```diff
@@ -0,0 +1,461 @@
+
+### Links
+
+A link to an internal or external page must be defined by a `label` and a `link` property.
+
+```js
+starlight({
+	sidebar: [
+		// A link to the CSS & Styling guide.
+		{ label: 'CSS & Styling', link: '/guides/css-and-tailwind/' },
+		// An external link to the Astro website.
+		{ label: 'Astro', link: 'https://astro.build/' },
+	],
+});
+```
+
+With the above configuration, the following sidebar will be generated:
+
+<SidebarPreview
+	config={[
+		{ label: 'CSS & Styling', link: '/guides/css-and-tailwind/' },
+		{ label: 'Astro', link: 'https://astro.build/' },
+	]}
+/>
+
+### Groups
+
+Links can be grouped together under a heading. Groups can contain both links and other sub-groups.
+
+A group is defined by a `label` string and an `items` array containing other links and/or groups.
+
+```js
+starlight({
+	sidebar: [
+		// A group of links labelled "Guides".
+		{
+			label: 'Guides',
+			items: [
+				{ label: 'Components', link: '/guides/components/' },
+				{ label: 'Internationalization (i18n)', link: '/guides/i18n/' },
+				// A nested group of links.
+				{
+					label: 'Styling',
+					items: [
+						{ label: 'CSS', link: '/guides/css-and-tailwind/' },
+						{ label: 'Tailwind', link: '/guides/css-and-tailwind/' },
+						{ label: 'Shiki', link: '/guides/css-and-tailwind/' },
+					],
+				},
+			],
+		},
+	],
+});
+```
+
+The above configuration will generate the following sidebar:
+
+<SidebarPreview
+	config={[
+		{
+			label: 'Guides',
+			items: [
+				{ label: 'Components', link: '/guides/components/' },
+				{ label: 'Internationalization (i18n)', link: '/guides/i18n/' },
+				{
+					label: 'Styling',
+					items: [
+						{ label: 'CSS', link: '/guides/css-and-tailwind/' },
+						{ label: 'Tailwind', link: '/guides/css-and-tailwind/' },
+						{ label: 'Shiki', link: '/guides/css-and-tailwind/' },
+					],
+				},
+			],
+		},
+	]}
+/>
+
+By combining links and groups, you can create a wide variety of sidebars layouts.
+
+## Badges
+
+Links can also include a `badge` property to display a badge next to the link label.
+
+```js
+starlight({
+	sidebar: [
+		{
+			label: 'Guides',
+			items: [
+				// A link with a "New" badge.
+				{
+					label: 'Components',
+					link: '/guides/components/',
+					badge: 'New',
+				},
+			],
+		},
+	],
+});
+```
+
+The above configuration will generate the following sidebar:
+
+<SidebarPreview
+	config={[
+		{
+			label: 'Guides',
+			items: [
+				{
+					label: 'Components',
+					link: '/guides/components/',
+					badge: { text: 'New', variant: 'default' },
+				},
+			],
+		},
+	]}
+/>
+
+The `badge` property includes a `text` property (the content to display in the badge) and optionally a `variant` property (to define the type of badge shown).
```

Comment:
> - Is it obvious enough that the badge property can accept a string or an object?

I think this needs clarification here. It’s a bit odd to flow straight from the code example showing `badge: 'New'` to the statement that it can have a `text` property.

Maybe we want a subsection for badge variants? That way we can frame the use case: “Want a variant? Use the object form”. That would also give us space to address your other point of where we can say that the default badge uses the site accent color.

Reply:
I think this is a good idea. I've added a new section for badge variants which contains the object example and the description of the default badge color and I "tried" to use the wording of some of your previous suggestions.